test/gateway_composer/transfer_token.js

```diff
@@ -0,0 +1,134 @@
+            const {
+                gatewayComposer,
+                valueToken,
+            } = await gatewayComposerUtils.setupGatewayComposer(accountProvider);
+
+            const to = accountProvider.get();
+            const amount = new BN(10);
+            utils.expectRevert(gatewayComposer.transferToken(
+                valueToken.address,
+                to,
+                amount,
+                { from: accountProvider.get() },
+            ),
+            'Should revert as msg.sender is not the owner.',
+            'Only owner can call the function.');
+        });
+
+        it('Should fail when EIP20 token address is invalid.', async () => {
+            const {
+                gatewayComposer,
+                owner,
+            } = await gatewayComposerUtils.setupGatewayComposer(accountProvider);
+
+            const to = accountProvider.get();
+            const amount = new BN(10);
+            utils.expectRevert(gatewayComposer.transferToken(
+                utils.NULL_ADDRESS,
+                to,
+                amount,
+                { from: owner },
+            ),
+            'Should revert as token address is null.',
+            'EIP20 token address is null.');
+        });
+
+        it('Should fail when owner balance is not sufficient.', async () => {
+            const {
+                gatewayComposer,
+                valueToken,
+                owner,
+            } = await gatewayComposerUtils.setupGatewayComposer(accountProvider);
+
+            // Set GC address balance to 0.
+            await valueToken.setBalance(gatewayComposer.address, new BN(0));
+
+            const to = accountProvider.get();
+            const amount = new BN(10);
+            utils.expectRevert(gatewayComposer.transferToken(
+                valueToken.address,
+                to,
+                amount,
+                { from: owner },
+            ),
+            "Should revert as owner doesn't have sufficient balance.");
+        });
+    });
+
+    describe('Positive Tests', async () => {
+        const accountProvider = new AccountProvider(accounts);
+
+        it('Returns true on successful execution.', async () => {
+            const {
+                gatewayComposer,
+                valueToken,
+                owner,
+            } = await gatewayComposerUtils.setupGatewayComposer(accountProvider);
+
+            const amount = new BN(10);
+            // Set GC address balance.
+            await valueToken.setBalance(gatewayComposer.address, amount);
+            const to = accountProvider.get();
+            const executionStatus = await gatewayComposer.transferToken.call(
+                valueToken.address,
+                to,
+                amount,
+                { from: owner },
+            );
+            assert.strictEqual(executionStatus, true);
+        });
+
+        it('Validates destination address balance after calling transferToken.', async () => {
```

Comment:
I do not think this test should be here.

This is a test of what will be true about an arbitrary address for which we have no information and over which we have no control.

Reply:
I think the assert is important to verify the balance after transfer to make sure if transferToken is actually working or not. One approach could be merge assert with above test "Returns true on successful execution." with above test instead of writing a separate test. What do you think?

Reply:
Verifying the balance requires interacting with an address that cannot be trusted.

We need to test how our contracts respond to inputs (e.g., the response from `transfer`). I do not see a value in testing how an untrustworthy/unknowable source responds to inputs (e.g., calling `transfer` or `balanceOf`).